crates/yttrium/src/chain_abstraction/send_transaction.rs

```diff
@@ -0,0 +1,112 @@
+use {
+    super::{api::FeeEstimatedTransaction, error::SendTransactionError},
+    crate::{
+        provider_pool::ProviderPool,
+        serde::{duration_millis, option_duration_millis, systemtime_millis},
+    },
+    alloy::{
+        consensus::{SignableTransaction, TxEnvelope},
+        primitives::{PrimitiveSignature, B256},
+        rpc::types::TransactionReceipt,
+    },
+    alloy_provider::Provider,
+    serde::{Deserialize, Serialize},
+    std::time::{Duration, Instant, SystemTime},
+};
+
+pub async fn send_transaction(
+    txn: FeeEstimatedTransaction,
+    sig: PrimitiveSignature,
+    provider_pool: &ProviderPool,
+) -> Result<
+    (TransactionReceipt, TransactionAnalytics),
+    (SendTransactionError, TransactionAnalytics),
+> {
+    let start = Instant::now();
+    let start_time = SystemTime::now();
+
+    let provider = provider_pool.get_provider(&txn.chain_id).await;
+    let signed = txn.into_eip1559().into_signed(sig);
+    let txn_hash = *signed.hash();
+
+    let send_start = Instant::now();
+    let sent_transaction_result =
+        provider.send_tx_envelope(TxEnvelope::Eip1559(signed)).await;
+    let send_latency = send_start.elapsed();
+    let sent_transaction = sent_transaction_result.map_err(|e| {
+        (
+            SendTransactionError::Rpc(e),
+            TransactionAnalytics {
```

Comment:
- do we track when the route transaction fails with the reason?
- do we track when the init transaction fails with the reason?
- if the whole flow was successful?

Reply:
Yes to all of those. LMK if the code looks otherwise